crafting-dead-core/src/main/java/com/craftingdead/core/commands/impl/CommandThirst.java

```diff
@@ -0,0 +1,131 @@
+/**
+ * Crafting Dead
+ * Copyright (C) 2020  Nexus Node
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.craftingdead.core.commands.impl;
+
+import com.craftingdead.core.commands.CommandBase;
+import com.craftingdead.core.potion.ModEffects;
+import com.mojang.brigadier.CommandDispatcher;
+import com.mojang.brigadier.arguments.StringArgumentType;
+import com.mojang.brigadier.exceptions.CommandSyntaxException;
+import net.minecraft.command.CommandSource;
+import net.minecraft.command.Commands;
+import net.minecraft.command.ISuggestionProvider;
+import net.minecraft.entity.player.ServerPlayerEntity;
+import net.minecraft.potion.EffectInstance;
+import net.minecraft.util.SoundCategory;
+import net.minecraft.util.SoundEvents;
+import net.minecraft.util.text.ITextComponent;
+import net.minecraft.util.text.StringTextComponent;
+import net.minecraft.util.text.TextFormatting;
+import net.minecraft.util.text.TranslationTextComponent;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * This class is a example of command using custom CommandBase class ({@link CommandBase});
+ * Function: Cure Thirst
+ */
+public class CommandThirst extends CommandBase {
+
+  private static final ITextComponent SUCCESS_MESSAGE_OPERATOR;
+  private static final ITextComponent SUCCESS_MESSAGE_CONFIRMATION;
+  private static final ITextComponent FAILURE_MESSAGE_NOT_USED;
+
+  public CommandThirst() {
+    super("thirst", null, 3);
+  }
+
+  @Override
+  public void register(CommandDispatcher<CommandSource> dispatcher) {
```

Comment:
This can just be a static method, no need to register instances of `CommandBase` in a map. See how vanilla does commands - I think this whole system is a bit overengineered. All that's necessary is a static register method in a dedicated class for each command.